walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,89 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.ArrayList;
+
+import java.util.Comparator;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
+  /*
+  * For a particular time slot to work, all attendees must be free to attend the meeting. When a query is made, it will be given a collection of all known events. 
+  * SOLUTION : Check to see if any attendees from the meeting request are already in events for a specfied time, if they are then the timeslot is not avaiable else it is.
+  * Return 
+  */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    /*
+    // sort by start time inplace
+    EventComparator eventComparator = new EventComparator();
+    Collections.sort(events, eventComparator);
```

Comment:
To do this, you could create a new ArrayList that is a copy of the events, and then use Collections.sort to sort it in place.